File: P1/Ejercicio3/margarita_builder.py
from pizza_builder import PizzaBuilder
from pizza import Pizza

class MargaritaBuilder(PizzaBuilder):
    def __init__(self):
        self.reset()
    
    def reset(self):
        self._pizza = Pizza()

    def add_ingredientes(self):
        self._pizza.ingredientes = "mozzarella, albahaca"
    
    def add_salsa(self):
        self._pizza.salsa = "tomate"

    def set_tamano(self):
        self._pizza.tamano = "mediana"
    
    def get_result(self):
        pizza = self._pizza
        self.reset()
        return pizza

File: P4/ejercicio_grupal/lib/builders/margarita_builder.dart
import '/builders/pizza_builder.dart';

class MargaritaBuilder extends PizzaBuilder {
  final String nombre = 'Margarita';
  final double precio = 8.50;
  final List<String> ingredientes = ['Tomate', 'Mozzarella', 'Albahaca'];

  @override
  void setNombre() {
    pizza.setNombre = nombre;
  }

  @override
  void setPrecio() {
    pizza.setPrecio = precio;
  }

  @override
  void setTamano(String tamano) {
    pizza.setTamano = tamano;
  }

  @override
  void setIngredientes() {
    pizza.setIngredientes = ingredientes;
  }
}
